tools/bin/ecflow/processes/main/post_processing/tm_analysis.py

```diff
@@ -102,35 +102,63 @@
             # relative to historic range
             f = interp1d(x, q)
             percentile = f(value)
-            combine = (lat, lon, float(percentile))
+            
+            if percentile < 2:
+                category = 0
+            elif 2 <= percentile < 5:
+                category = 1
+            elif 5 <= percentile < 10:
+                category = 2
+            elif 10 <= percentile < 20:
+                category = 3
+            elif 20 <= percentile < 30:
+                category = 4
+            elif 30 <= percentile < 70:
+                category = 5
+            elif 70 <= percentile < 80:
+                category = 6
+            elif 80 <= percentile < 90:
+                category = 7
+            elif 90 <= percentile < 95:
+                category = 8
+            elif 95 <= percentile < 98:
+                category = 9
+            elif percentile >= 98:
+                category = 10
+            
+            combine = (lat, lon, float(percentile), category)
             d.append(combine)
 
             # if interpolation fails then a value is assigned based on
             # whether it is higher or lower than the range
         except ValueError:
             if (value > max(x)):
                 percentile = (max_q)
-                combine = (lat, lon, percentile)
+                category = 10
+                combine = (lat, lon, percentile, category)
                 d.append(combine)
             else:
                 percentile = (min_q)
-                combine = (lat, lon, percentile)
+                category = 0
+                combine = (lat, lon, percentile, category)
                 d.append(combine)
     except IOError:
-        percentile = value
-        combine = (lat, lon)
+        combine = (lat, lon, -9999.0, -9999.0)
         d.append(combine)
 
 # Dictionary to DataFrame to Dataset
-df = pd.DataFrame(d, columns=["Latitude", "Longitude", "tmpercentile"])
+df = pd.DataFrame(d, columns=["Latitude", "Longitude", "tmpercentile", "category"])
```

Comment:
try to keep quotation marks consistent -- I like to use `'` for everything, but if you want to use `"`, use it for everything in the same script.

Reply:
Good point - fixed!